src/ChainSafe.GamingSDK.EVM/Web3/Core/Environment/ILogWriter.cs

```diff
@@ -0,0 +1,8 @@
+﻿namespace ChainSafe.GamingWeb3.Environment
+{
+  public interface ILogWriter
```

Comment:
Not sure why an enum is a better solution than separate methods

Reply:
The real benefit comes when you want to implement log filtering. You can do `logger.IsLevelEnabled(LogLevel.Info)` instead of having 5 more methods. It also becomes really easy to implement the actual filtering itself. Anyway, I'm not suggesting we do this right now, I'm suggesting we add a TODO comment.

Reply:
We can use your approach but add extension methods for ILogWriter like _logger.LogWarning(msg) => _logger.Log(LogLevel.Warning, msg). This way you'd have to code just 1 method for each logger implementation, but they all would support shorter method calls thanks to extensions methods.

Reply:
Also makes sense.